appeals/web/src/server/lib/middleware/add-apiclient-to-request.js

```diff
@@ -0,0 +1,41 @@
+import { createHttpLoggerHooks } from '@pins/platform';
+import config from '@pins/appeals.web/environment/config.js';
+import got from 'got';
+import pino from '../logger.js';
+import * as authSession from '../../app/auth/auth-session.service.js';
+
+const [requestLogger, responseLogger] = createHttpLoggerHooks(pino, config.logLevelStdOut);
+
+const instance = got.extend({
+	prefixUrl: config.apiUrl,
+	responseType: 'json',
+	resolveBodyOnly: true
+});
+
+const getInstance = (/** @type {string} */ userId) =>
+	instance.extend({
+		hooks: {
+			beforeRequest: [
+				requestLogger,
+				async (options) => {
+					options.headers.userid = userId;
```

Comment:
Are we going with header? I thought we settled on body param, but either way proves the principle.

Reply:
I think David and I would prefer to go with the header, keeping separate...

Reply:
I think we discussed calling `userid` something less generic so it doesn't conflict with other uses of `userid` and it's a bit clearer that it doesn't refer to a value in the BO DB.

Reply:
OK no worries, header will be fine. - yeah something more specific like `authUserId` or `azureAdUserId`